### PR TITLE
fixing replace logic

### PR DESCRIPTION
**Pull request by spunkedy**
The if condition didn't take account for allowing the current document to be replaced.

screen shot 2016-03-27 at 2 26 21 pm

example mp4

### CODE REVIEW

File: src/navigation.js [hunk 1]
```diff
@@ -222,7 +222,9 @@ function cleanNavigate(doc, replace = false) {
     let docs = navigationDocument.documents;
     let last = getLastDocumentFromStack();
 
-    if (!last || last !== loaderDoc || last !== errorDoc || !replace) {
+    console.log("trying to clean navigate inside cleanNavigate func");
```
Comment (emadalam) [inline src/navigation.js]: @spunkedy Can you please remove the `console.log` statement and commit the changes. Thanks a lot for the gotcha :+1:

File: src/navigation.js [hunk 2]
```diff
@@ -222,7 +222,9 @@ function cleanNavigate(doc, replace = false) {
...
+    if (!replace && (!last || last !== loaderDoc || last !== errorDoc )) {
+        console.log("pushed instead of replaced");
```
Comment (emadalam) [inline src/navigation.js]: @spunkedy Same as above :+1:

File: src/navigation.js [hunk 3]
```diff
@@ -317,6 +319,7 @@ function navigate(page, options, replace) {
                     _.defer(presentModal, doc);
                 } else { // navigate
                     // defer to avoid clashes with any ongoing process (tvmlkit weird behavior -_-)
+                    console.log("trying to clean navigate");
```
Comment (emadalam) [inline src/navigation.js]: @spunkedy Same as above :+1:

**Comment by spunkedy** [comment]
The debug logging should be out now.

**Pull request merged**